### PR TITLE
fix: [CDS-54040]: error handling for FileNotFoundException while rend…

### PR DESCRIPTION
**Pull request by achyuth-harness**
…ering helm template

## Describe your changes
In case directory path was incorrect (in this case sub-chart name was whitespace), we were not doing the error handling correctly. We have now handled this case.
Error message before fix:
image

Current error message:
image

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44771)

### CODE REVIEW

**Review by GoelPratyush99 — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44771)* status: 2 file(s) to review. 1 of 2 approvals obtained.

File: 930-delegate-tasks/src/main/java/io/harness/delegate/exceptionhandler/handler/HelmClientRuntimeExceptionHandler.java [hunk 1]
```diff
@@ -145,6 +145,10 @@ private WingsException handleRenderChartException(HelmClientException helmClient
     if (lowerCaseMessage.contains("invalid kubernetes yaml")) {
       return NestedExceptionUtils.hintWithExplanationException(
           HINT_INVALID_YAML, EXPLAIN_INVALID_YAML, new InvalidRequestException(message));
+    } else if (lowerCaseMessage.contains("chart.yaml file is missing")) {
```
Comment (abosii-harness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/exceptionhandler/handler/HelmClientRuntimeExceptionHandler.java:148]: We need to handle the case when user doesn't provide an existing subchart

**Review by abosii-harness — CHANGES_REQUESTED** [review]

**Review by achyuth-harness — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44771#-:-NQ0dgug2YuFBEnR_wkl:bnfp4nl)* status: 2 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/exceptionhandler/handler/HelmClientRuntimeExceptionHandler.java` line 148 at r2](https://reviewable.io/reviews/harness/harness-core/44771#-NPrv2ce17nyAZtL0E8e-r2-148:-NQ0dedSAuX7VjSNm37K:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/8e5c01822c936f3bd041cdf53348b3f18ca06f50/930-delegate-tasks/src/main/java/io/harness/delegate/exceptionhandler/handler/HelmClientRuntimeExceptionHandler.java#L148)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

We need to handle the case when user doesn't provide an existing subchart
</blockquote></details>

Done.

**Review by abosii-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44771)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Review by GoelPratyush99 — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44771)* status: :shipit: complete! ✓

**Pull request merged**